pkg/webhooks/workload_webhook.go

```diff
@@ -45,40 +42,10 @@ type WorkloadWebhook struct{}
 func setupWebhookForWorkload(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(&kueue.Workload{}).
-		WithDefaulter(&WorkloadWebhook{}).
 		WithValidator(&WorkloadWebhook{}).
 		Complete()
 }
 
-// +kubebuilder:webhook:path=/mutate-kueue-x-k8s-io-v1beta1-workload,mutating=true,failurePolicy=fail,sideEffects=None,groups=kueue.x-k8s.io,resources=workloads,verbs=create,versions=v1beta1,name=mworkload.kb.io,admissionReviewVersions=v1
-
-var _ webhook.CustomDefaulter = &WorkloadWebhook{}
-
-// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
-func (w *WorkloadWebhook) Default(ctx context.Context, obj runtime.Object) error {
```

Comment:
As for podSet name, we added the marker `+kubebuilder:default` so that we can replace the defaulter https://github.com/kubernetes-sigs/kueue/blob/6176fd1799fbd42ea7ee112703a15539b4d0b785/apis/kueue/v1beta1/workload_types.go#L121 
Also the WorkloadSpec includes the marker `// +kubebuilder:validation:MinItems=1` for PodSet https://github.com/kubernetes-sigs/kueue/blob/6176fd1799fbd42ea7ee112703a15539b4d0b785/apis/kueue/v1beta1/workload_types.go#L35C2-L35C39

And for `MinCount` the value will be ignored while the `PartialAdmission` feature is disabled, right?

Reply:
ok for `name`.

For `minCount`, the standard practice is to actually drop the values when the feature is disabled, regardless of whether they are ignored.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#:~:text=if%20!utilfeature.DefaultFeatureGate.Enabled(features.Frobber2D)%20%26%26%20oldFrobber.Width%20%3D%3D%20nil%20%7B

However, maybe it's just time to graduate `PartialAdmission` to stable and get rid of the feature gate. But that would have to be a separate PR.

Reply:
Defaulting for minCount restored